bundles/org.openhab.core/src/test/java/org/eclipse/smarthome/core/library/types/DateTimeTypeTest.java

```diff
@@ -104,13 +136,48 @@ public ParameterSet(TimeZone defaultTimeZone, Map<String, Integer> inputTimeMap,
          * @param defaultTimeZone
          * @param inputTimeString
          * @param expectedResult
+         * @param expectedResultLocalTZ
          */
-        public ParameterSet(TimeZone defaultTimeZone, String inputTimeString, String expectedResult) {
+        public ParameterSet(TimeZone defaultTimeZone, String inputTimeString, String expectedResult,
+                String expectedResultLocalTZ) {
             this.defaultTimeZone = defaultTimeZone;
             this.inputTimeMap = null;
             this.inputTimeZone = null;
             this.inputTimeString = inputTimeString;
             this.expectedResult = expectedResult;
+            this.expectedResultLocalTZ = expectedResultLocalTZ;
+            this.locale = null;
+            this.pattern = null;
+            this.expectedFormattedResult = expectedResult.substring(0, 19);
+        }
+
+        /**
+         * create a parameter set with either {@link ParameterSet#inputTimeMap} and {@link ParameterSet#inputTimeZone}
+         * parameters or {@link ParameterSet#inputTimeString} parameter. Non default values can be set for testing
+         * the format method.
+         *
+         * @param defaultTimeZone
+         * @param inputTimeMap
+         * @param inputTimeZone
+         * @param inputTimeString
+         * @param expectedResult
+         * @param expectedResultLocalTZ
+         * @param locale
+         * @param pattern
+         * @param expectedFormattedResult
+         */
+        public ParameterSet(TimeZone defaultTimeZone, Map<String, Integer> inputTimeMap, TimeZone inputTimeZone,
```

Comment:
I am just wondering if we can reuse this constructor called by the other constructors. But that is only a side comment. Up to you.

Reply:
Yes we can.

Reply:
Done